app/code/community/Bolt/Boltpay/Model/Observer.php

```diff
@@ -157,6 +158,55 @@ public function addMessageWhenCapture($observer)
         }
     }
 
+    /**
+     * Updates the Bolt transaction status on order status change.
+     *
+     * @param $observer
+     */
+    public function updateBoltTransactionStatus($observer)
+    {
+        /* @var Mage_Sales_Model_Order $order */
+        $order = $observer->getOrder();
+
+        /** @var Mage_Sales_Model_Order_Payment $payment */
+        $payment = $order->getPayment();
+
+        $method = $payment->getMethod();
+        if (strtolower($method) == Bolt_Boltpay_Model_Payment::METHOD_CODE) {
```

Comment:
@daisy1754 I think Leon just try update the bolt transaction statut for order on merchant server, not bolt server

Reply:
`boltTransactionStatus` is a legacy field that appeared to save the transaction status at only order creation time.  More importantly, this field has been exclusively used to determine what are valid **Magento** order transitions.  So, the code is mixing the ideas Bolt order status and Magento order status and lacked mappings for cases many cases like 'on-hold' which is not a 
Bolt Status, and other subtle ones like deferred.

Important things are though it is not semantically correct, the mappings are functionally correct with how the system uses this field.  If we wish for it to be semantically correct, then we drop this field and simply rely on Magento order state, removing all `Payment::TRANSACTION_*` constants. and updating `Payment::$_validStateTransitions`, and methods `Payment::translateHookTypeToTransactionStatus`, and `Payment::handleTransactionUpdate` at minimum.  Because this seemed to me to be a very heavy-handed approach compared to the simpler and stable solution presented, I evaluated the risk of such a change for semantic value out-weighing the benefits, I opted not to pursue this route.

However, both approaches will function the same.  Perhaps renaming the field can be the happy medium if this naming presents a problem with code comprehension.